<a href="https://colab.research.google.com/github/Dataly8/bike_sharing_estadistica/blob/main/Analisis_uso_bicicletas_2012.ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 **El siguiente conjunto de datos (day.csv) contiene información acerca del uso de un servicio de alquiler de bicicletas.
 Para el año 2012, calcula la media de uso mensual (mes a mes de la columna “cnt”) e indica los 3 meses donde hubo más dispersión en el uso.**



In [16]:
import pandas as pd

In [17]:
public_csv_url ='https://raw.githubusercontent.com/Dataly8/bike_sharing_estadistica/main/day.csv'

In [18]:
df = pd.read_csv(public_csv_url)
df

,instant,dteday,season,yr,mnth,holiday,weekday,workingday,weathersit,temp,atemp,hum,windspeed,casual,registered,cnt
0,1,2011-01-01,1,0,1,0,6,0,2,0.344167,0.363625,0.805833,0.160446,331,654,985
1,2,2011-01-02,1,0,1,0,0,0,2,0.363478,0.353739,0.696087,0.248539,131,670,801
2,3,2011-01-03,1,0,1,0,1,1,1,0.196364,0.189405,0.437273,0.248309,120,1229,1349
3,4,2011-01-04,1,0,1,0,2,1,1,0.200000,0.212122,0.590435,0.160296,108,1454,1562
4,5,2011-01-05,1,0,1,0,3,1,1,0.226957,0.229270,0.436957,0.186900,82,1518,1600
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
726,727,2012-12-27,1,1,12,0,4,1,2,0.254167,0.226642,0.652917,0.350133,247,1867,2114
727,728,2012-12-28,1,1,12,0,5,1,2,0.253333,0.255046,0.590000,0.155471,644,2451,3095
728,729,2012-12-29,1,1,12,0,6,0,2,0.253333,0.242400,0.752917,0.124383,159,1182,1341
729,730,2012-12-30,1,1,12,0,0,0,1,0.255833,0.231700,0.483333,0.350754,364,1432,1796


Filtro por el año 2012 que se representa con un 1 en la columna 'yr'.
(yr: 0 = 2011, 1 = 2012)

In [19]:
df[df['yr'] == 1]

,instant,dteday,season,yr,mnth,holiday,weekday,workingday,weathersit,temp,atemp,hum,windspeed,casual,registered,cnt
365,366,2012-01-01,1,1,1,0,0,0,1,0.370000,0.375621,0.692500,0.192167,686,1608,2294
366,367,2012-01-02,1,1,1,1,1,0,1,0.273043,0.252304,0.381304,0.329665,244,1707,1951
367,368,2012-01-03,1,1,1,0,2,1,1,0.150000,0.126275,0.441250,0.365671,89,2147,2236
368,369,2012-01-04,1,1,1,0,3,1,2,0.107500,0.119337,0.414583,0.184700,95,2273,2368
369,370,2012-01-05,1,1,1,0,4,1,1,0.265833,0.278412,0.524167,0.129987,140,3132,3272
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
726,727,2012-12-27,1,1,12,0,4,1,2,0.254167,0.226642,0.652917,0.350133,247,1867,2114
727,728,2012-12-28,1,1,12,0,5,1,2,0.253333,0.255046,0.590000,0.155471,644,2451,3095
728,729,2012-12-29,1,1,12,0,6,0,2,0.253333,0.242400,0.752917,0.124383,159,1182,1341
729,730,2012-12-30,1,1,12,0,0,0,1,0.255833,0.231700,0.483333,0.350754,364,1432,1796


Agrupo por 'mnth' y promediamos la columna 'cnt'

In [26]:
df[df['yr'] == 1].groupby('mnth')['cnt'].mean().round(2)

,cnt
mnth,
1,3120.77
2,3556.45
3,5318.55
4,5807.47
5,6318.23
6,6761.00
7,6567.97
8,6919.45
9,7285.77


Ejecuto la desviación estandar con .std() que por defecto siempre hace el cálculo de la muestral (divide por n-1).

In [27]:
df[df['yr'] == 1].groupby('mnth')['cnt'].std().round(2)

,cnt
mnth,
1,872.85
2,870.72
3,1251.16
4,1308.94
5,1078.40
6,954.20
7,867.18
8,794.06
9,979.16


Cálculo del coeficiente de variación para entender la variación relativa de cada mes y comparar con la desviación estándar.

In [29]:
media = df[df['yr'] == 1].groupby('mnth')['cnt'].mean().round(2)
std = df[df['yr'] == 1].groupby('mnth')['cnt'].std().round(2)
cv = std/media
cv.round(4)

,cnt
mnth,
1,0.2797
2,0.2448
3,0.2352
4,0.2254
5,0.1707
6,0.1411
7,0.1320
8,0.1148
9,0.1344


Lo junto en una tabla comparativa:

In [37]:
tabla = pd.DataFrame({'Media': media, 'Desv.Estándar': std, 'Coef.Variación' : cv})

In [38]:
tabla.round(2)

,Media,Desv.Estándar,Coef.Variación
mnth,,,
1,3120.77,872.85,0.28
2,3556.45,870.72,0.24
3,5318.55,1251.16,0.24
4,5807.47,1308.94,0.23
5,6318.23,1078.40,0.17
6,6761.00,954.20,0.14
7,6567.97,867.18,0.13
8,6919.45,794.06,0.11
9,7285.77,979.16,0.13


### **Cálculo de la media mes a mes**:

- El uso medio de alquiler de bicicletas **crece en los meses más cálidos, y presenta un pico en el mes de septiembre con 7285.77 bicis al día**.

- La media **desciende considerablemente en diciembre y enero**.

Estos datos son coherente dado que se trata de un servicio al aire libre, así que lo esperado es que el alquiler y uso suban en los meses más soleados.

### **Medidas de dispersión:**

Para entender la dispersión de los datos, se han calculado dos medidas complementarias:



La **desviación estándar** (σ) que expresa la dispersión en unidades absolutas.

  *Meses con mayor ranking:*

    1.   Octubre (1941.82)
    2.   Diciembre (1803.23)
    3.   Abril (1308.94)


El **coeficiente de variación** que expresa expresa la dispersión en relación al nivel medio de uso de cada mes.

  *Meses con mayor ranking:*

    1.   Diciembre (45%)
    2.   Octubre (30%)
    3.   Enero (28%)


**Interpretación de los resultados:**

Octubre y Diciembre son los meses que aparecen con mayor dispersión, tanto en desviación estándar como en coefiente de correlación. La diferencia está en el tercer puesto: Abril entra en el ranking de desv. estándar, en cambio, Enero entra por coef. de correlación.

Esto se puede explicar porque la medida de coeficiente de correlación normaliza por el nivel de uso: Diciembre y Enero son meses de baja demanda y su variación es muy grande en proporción a su media.

Se puede establecer una hipótesis: los meses de mayor dispersión coinciden con periodos de transición estacional (en otoño y en invierno).
La inestabilidad del clima podría generar días de uso muy dispares.


Esta hipótesis y análisis se podrían ampliar y verificar cruzando los datos de 'cnt' con las variables meteorológicas del dataset ('weathersit', 'temp', 'windspeed') y también con el tipo de día ('workingday')

